usecase/patientData.go

```diff
@@ -434,17 +434,17 @@ func writeFromIterV1(ctx context.Context, res *bytes.Buffer, bgUnit string, p *w
 			if bgUnit != "" {
 				switch datum["type"] {
 				case "smbg":
-					if datum["units"] != bgUnit {
+					if datum["units"] != bgUnit && isConvertibleUnit(datum["units"].(string)) {
 						if datum["units"] == MmolL {
 							datum["units"] = MgdL
-							datum["value"] = getMgdl(datum["value"].(float64))
+							datum["value"] = convertToMgdl(datum["value"].(float64))
 						} else {
 							datum["units"] = MmolL
-							datum["value"] = getMmol(datum["value"].(float64))
+							datum["value"] = convertToMmol(datum["value"].(float64))
 						}
 					}
 				case "wizard":
-					/*For wizard, we don't have anymore fields in mmol, so we're changing the unit but no conversion is done.
+					/*For wizard, we don't have anymore fields in mmol, so we're changing the given but no conversion is done.
```

Comment:
less impact but looks like the unit -> given replace may be not appropriate here